### ISSUE TITLE
Native linking failed, undefined Objective-C class after updating from 11.9.1.24 to 11.10.1.177

### ISSUE THREAD

**Issue by therealjohn** [xamarin/xamarin-macios#4301]
_From Developer Community_
In one of our developers' environments Visual Studio (for Windows) updated Xamarin.iOS from 11.9.1.24 to 11.10.1.177.  This has broken the build of one of our iOS apps with several "Native linking failed, Undefined Objective-C class..." errors.  We have verified this by successfully building the application in another environment using 11.9.1.24.  Is there a direct link for Xamarin.VisualStudio 4.9.0.753 installer?  Or an alternate method to downgrade Xamarin extensions? Downgrading the entire Visual Studio environment is not acceptable as these environments have been meticulously setup and we cannot afford to rebuild the entire VS environment.  Thank you.

### Expected Behavior
No build errors.

### Actual Behavior
```
Native linking failed, undefined Objective-C class: FileUtil. The symbol '_OBJC_CLASS_$_FileUtil' could not be found in any of the libraries or frameworks linked with your application.
Native linking failed, undefined Objective-C class: FormatUtil. The symbol '_OBJC_CLASS_$_FormatUtil' could not be found in any of the libraries or frameworks linked with your application.
Native linking failed, undefined Objective-C class: GraphicsUtil. The symbol '_OBJC_CLASS_$_GraphicsUtil' could not be found in any of the libraries or frameworks linked with your application.
Native linking failed, undefined Objective-C class: MagCardReader. The symbol '_OBJC_CLASS_$_MagCardReader' could not be found in any of the libraries or frameworks linked with your application.
Native linking failed, undefined Objective-C class: SmartCardReader. The symbol '_OBJC_CLASS_$_SmartCardReader' could not be found in any of the libraries or frameworks linked with your application.
Native linking failed, undefined Objective-C class: ToolsUtil. The symbol '_OBJC_CLASS_$_ToolsUtil' could not be found in any of the libraries or frameworks linked with your application.
Native linking failed, undefined Objective-C class: ZebraPrinter. The symbol '_OBJC_CLASS_$_ZebraPrinter' could not be found in any of the libraries or frameworks linked with your application.
Native linking failed, undefined Objective-C class: ZebraPrinterConnection. The symbol '_OBJC_CLASS_$_ZebraPrinterConnection' could not be found in any of the libraries or frameworks linked with your application.
```
### Environment

VS 2017 15.6.5

### Build Logs

Please ping me. They are provided privately on https://developercommunity.visualstudio.com/content/problem/264659/xamarinios-1110-breaks-ios-build-with-errors-nativ.html?childToView=277473#comment-277473

### Additional Info

I compared the build logs and didn't see any difference in iOS build or configuration settings. I was not able to adequately compare the `mtouch` parameters though.

**Comment by rolfbjarne** [comment]
This might be a duplicate of https://github.com/xamarin/xamarin-macios/issues/4065.

@therealjohn can you check if this is the case? The problem is with binding assemblies compiled a long time ago; that issue contains a tool that can be executed on those assemblies to fix them.

**Comment by therealjohn** [comment]
@rolfbjarne Thanks! I can ask the original user and find out.

**Comment by PaulMatthewMadden** [comment]
@rolfbjarne We do not have the source code of the wrapper in question (I am attempting to get it from a 3rd party developer).  In the meantime I applied the fix at https://github.com/rolfbjarne/fix-binding-library to the dll.  I also added "--registrar:static" as additional mtouch arguments.  We are now receiving the following error:

`Native linking failed, duplicate symbol`

I am attaching a diagnostic build log.  Thank you for any assistance you can provide!

[Native Linking Error - 2 Duplicate Symbols.zip](https://github.com/xamarin/xamarin-macios/files/2134420/Native.Linking.Error.-.2.Duplicate.Symbols.zip)

**Comment by rolfbjarne** [comment]
@PaulMatthewMadden it seems there's a problem with the bindings for the Estimote library. It looks like this is already fixed in the latest version of the source code, so I've contacted our components team to see if we can just push an updated nuget.

**Comment by rolfbjarne** [comment]
@PaulMatthewMadden the Estimote nuget has been updated, can you try the updated version and see if that fixes the problem for you?

**Comment by mandel-macaque** [comment]
@PaulMatthewMadden I will close the issues since we have not received any updates. Please try with the new nuget as @rolfbjarne  mentioned and if it does not fix the issue, please feel free to reopen the issue.

**Comment by PaulMatthewMadden** [comment]
Apologies for the late response.  This has corrected the issue!  Thank you.

The exact steps I used to correct the error(s):

1.) I used rolfbjarne's "fix-binding-library" on the offending DLL that caused the original errors.  In this case the offending DLL was iOS.Zebralink.dll.

2.) I updated the Nuget package "Xamarin.Estimote.iOS".

Note:  The additional mtouch argument "--registrar:static" was not required in my situation.

Thank you!
